modules/rostests/apitests/gdi32/StretchBlt.c

```diff
@@ -18,91 +18,857 @@
+
+    if ((SrcTopDown || DstTopDown) && !(SrcTopDown && DstTopDown))
+    {
+        expected[0] = 0x0000FF00;
+        expected[1] = 0x000000FF;
+        expected[2] = 0xFF000000;
+        expected[3] = 0x00FF0000;
+    }
+    else
+    {
+        expected[0] = 0xFF000000;
+        expected[1] = 0x00FF0000;
+        expected[2] = 0x0000FF00;
+        expected[3] = 0x000000FF;
+    }
+
+    test_StretchBlt_stretch(hdcDst, hdcSrc, &biDst, dstBuffer, srcBuffer,
+                             1, 1, -2, -2, 0, 0, 2, 2, expected, __LINE__, SrcTopDown, DstTopDown); // Case 0011 (3) - Both Flip.
+
+    if ((SrcTopDown || DstTopDown) && !(SrcTopDown && DstTopDown))
+    {
+        expected[0] = 0xFF000000;
+        expected[1] = 0x00FF0000;
+        expected[2] = 0x0000FF00;
+        expected[3] = 0x000000FF;
+    }
+    else
+    {
+        expected[0] = 0x0000FF00;
+        expected[1] = 0x000000FF;
+        expected[2] = 0xFF000000;
+        expected[3] = 0x00FF0000;
+    }
+
+    test_StretchBlt_stretch(hdcDst, hdcSrc, &biDst, dstBuffer, srcBuffer,
+                             1, 1, -2, -2, 0, 1, 2, -2, expected, __LINE__, SrcTopDown, DstTopDown); // Case 0010 (2) - Horizontal Flip. 
+
+    if ((SrcTopDown || DstTopDown) && !(SrcTopDown && DstTopDown))
+    {
+        expected[0] = 0x000000FF;
+        expected[1] = 0x0000FF00;
+        expected[2] = 0x00FF0000;
+        expected[3] = 0xFF000000;
+    }
+    else
+    {
+        expected[0] = 0x00FF0000;
+        expected[1] = 0xFF000000;
+        expected[2] = 0x000000FF;
+        expected[3] = 0x0000FF00;
+    }
+
+    test_StretchBlt_stretch(hdcDst, hdcSrc, &biDst, dstBuffer, srcBuffer,
+                             1, 1, -2, -2, 1, 0, -2, 2, expected, __LINE__, SrcTopDown, DstTopDown); // Case 0001 (1) - Vertical Flip. 
+
+    if ((SrcTopDown || DstTopDown) && !(SrcTopDown && DstTopDown))
+    {
+        expected[0] = 0x00FF0000;
+        expected[1] = 0xFF000000;
+        expected[2] = 0x000000FF;
+        expected[3] = 0x0000FF00;
+
+    }
+    else
+    {
+        expected[0] = 0x000000FF;
+        expected[1] = 0x0000FF00;
+        expected[2] = 0x00FF0000;
+        expected[3] = 0xFF000000;
+    }
+
+    test_StretchBlt_stretch(hdcDst, hdcSrc, &biDst, dstBuffer, srcBuffer,
+                             2, 2, -2, -2, 2, 2, -2, -2, expected, __LINE__, SrcTopDown, DstTopDown); // Case 0000 (0) - No Flip. Special Case.
+
+    SelectObject(hdcSrc, oldSrc);
+    DeleteObject(bmpSrc);
+    DeleteDC(hdcSrc);
+
+    SelectObject(hdcDst, oldDst);
+    DeleteObject(bmpDst);
+    DeleteDC(hdcDst);
+
+    DeleteDC(hdcScreen);
+}
 
 START_TEST(StretchBlt)
 {
-    test_StretchBlt();
+    trace("\n\r\nStart of Generalized StretchBlt tests.\n\n");
+    test_StretchBlt();  // Without the delay in the next statement, the heading below is written in the middle of these tests fails.
+    Sleep(100);         // This allows the heading below to be written after this block of tests completes.
```

Comment:
Can you tell me why this waiting is needed?

Reply:
When using the DPRINT1's the header comments were misplaced as you can see here:
[StretchBlt_DPRINT1_Headers.txt](https://github.com/reactos/reactos/files/5981220/StretchBlt_DPRINT1_Headers.txt)

But after the change to "trace" statements, now it outputs correctly without any "sleep" statements as you can see here:
[AStretchBlt_trace_Headers.txt](https://github.com/reactos/reactos/files/5981227/AStretchBlt_trace_Headers.txt)

This is something that I did not expect to see changed after make the conversion from "DPRINT1" to "trace". This improved the debug output, so Thank You for the suggestion.